### ISSUE TITLE
salt remove.osd does not refresh grains on minion

### ISSUE THREAD

**Issue by bdekany** [SUSE/DeepSea#1376]
### Description of Issue/Question
`salt remove.osd` does not refresh grains on minion.
This cause an error when using other salt command which rely on `osd.list`.

### Setup
I had to remove an OSD from a minion without the intention of replacing it any time soon (aka I didn't run ceph.stage.3).
After a successful removal, the OSD id is still present in the grains of the minion.
I tried to launch a benchmark which failed due to the "phantom" OSD

### Steps to Reproduce Issue
List OSD
```shell
$ salt '*ses1*' osd.list
demo-ses1.zypp.local:
    - 1
    - 5
```
Remove OSD
```shell
$ salt-run disengage.safety
safety is now disabled for cluster ceph

$ salt-run remove.osd 1
True
True
```

OSD is still present in grains but not in ceph
```shell
$ salt '*ses1*' osd.list
demo-ses1.zypp.local:
    - 1
    - 5

$ ceph osd tree
ID CLASS WEIGHT  TYPE NAME          STATUS REWEIGHT PRI-AFF
-1       0.34084 root default
-5       0.04869     host demo-ses1
 5   hdd 0.04869         osd.5          up  1.00000 1.00000
```

Run a benchmark 
```shell
salt-run state.orch ceph.benchmarks.baseline
Running osd benchmarks........[ERROR   ] Exception occurred in runner benchmark.baseline: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 395, in _low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/srv/modules/runners/benchmark.py", line 284, in baseline
    parsed_results = [ast.literal_eval(r[master_minion]) for r in results]
  File "/usr/lib64/python2.7/ast.py", line 49, in literal_eval
    node_or_string = parse(node_or_string, mode='eval')
  File "/usr/lib64/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    Error ENOENT: problem getting command descriptions from osd.1
               ^
SyntaxError: invalid syntax
```

### Workaround
Refresh the grains after removal
```shell
$ salt '*ses1*' osd.retain
```

### Versions Report
Deepsea 0.8.5

Name        : salt-minion
Version     : 2016.11.4
Release     : 46.20.2
Architecture: x86_64
Install Date: Sun Aug  5 14:35:02 2018
Group       : System/Management
Size        : 38128
License     : Apache-2.0
Signature   : RSA/SHA256, Wed Apr 25 12:19:17 2018, Key ID 70af9e8139db7c82
Source RPM  : salt-2016.11.4-46.20.2.src.rpm
Build Date  : Wed Apr 25 12:16:10 2018
Build Host  : sheep16
Relocations : (not relocatable)
Packager    : https://www.suse.com/
Vendor      : SUSE LLC <https://www.suse.com/>
URL         : http://saltstack.org/
Summary     : The client component for Saltstack
Description :
Salt minion is queried and controlled from the master.
Listens to the salt master and execute the commands.
Distribution: SUSE Linux Enterprise 12

Name        : salt-master
Version     : 2016.11.4
Release     : 46.20.2
Architecture: x86_64
Install Date: Sun Aug  5 14:51:41 2018
Group       : System/Management
Size        : 1662872
License     : Apache-2.0
Signature   : RSA/SHA256, Wed Apr 25 12:19:17 2018, Key ID 70af9e8139db7c82
Source RPM  : salt-2016.11.4-46.20.2.src.rpm
Build Date  : Wed Apr 25 12:16:10 2018
Build Host  : sheep16
Relocations : (not relocatable)
Packager    : https://www.suse.com/
Vendor      : SUSE LLC <https://www.suse.com/>
URL         : http://saltstack.org/
Summary     : The management component of Saltstack with zmq protocol supported
Description :
The Salt master is the central server to which all minions connect.
Enabled commands to remote systems to be called in parallel rather
than serially.
Distribution: SUSE Linux Enterprise 12

**Comment by callithea** [comment]
Hi @bdekany I'm not sure if this helps but you could try running ``salt '*' saltutil.pillar_refresh`` on your salt-master and afterwards check with ``salt '*ses1*' osd.list`` if the osd is still present in DeepSea.

**Comment by bdekany** [comment]
Issue is fixed in deepsea 0.8.6
